lib/unified_health_data/service.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require_relative 'configuration'
+require_relative 'medical_record'
+
+module UnifiedHealthData
+  class Service < Common::Client::Base
+    configuration UnifiedHealthData::Configuration
+
+    def initialize(user)
+      super()
+      @user = user
+    end
+
+    def get_medical_records(start_date:, end_date:)
+      token = fetch_access_token
+      patient_id = @user.icn
+      path = "#{config.base_path}labs?patient-id=#{patient_id}&start-date=#{start_date}&end-date=#{end_date}"
+      response = perform(:get, path, nil, { 'Authorization' => token })
+      body = parse_response_body(response.body)
+
+      combined_records = fetch_combined_records(body)
+      parse_medical_records(combined_records)
+    end
+
+    private
+
+    def fetch_access_token
+      response = connection.post(config.token_path) do |req|
+        req.headers['Content-Type'] = 'application/json'
+        req.body = {
+          appId: config.app_id,
+          appToken: config.app_token,
+          subject: 'VA.gov SCDF Proxy Client',
+          userType: 'SYSTEM'
+        }.to_json
+      end
+      response.headers['authorization']
+    end
+
+    def parse_response_body(body)
+      # FIXME: workaround for testing
+      body.is_a?(String) ? JSON.parse(body) : body
+    end
+
+    def fetch_combined_records(body)
+      vista_records = body.dig('vista', 'entry') || []
+      oracle_health_records = body.dig('oracle-health', 'entry') || []
+      vista_records + oracle_health_records
+    end
+
+    def parse_medical_records(records)
+      records.select { |record| record['resource']['resourceType'] == 'DiagnosticReport' }.map do |record|
+        parse_single_record(record)
+      end
+    end
+
+    def parse_single_record(record)
+      location = fetch_location(record)
+      code = fetch_code(record)
+      sample_site = fetch_sample_site(record)
+      observations = fetch_observations(record)
+      ordered_by = fetch_ordered_by(record)
+
+      attributes = UnifiedHealthData::MedicalRecord::Attributes.new(
```

Comment:
Make non required attributes optional. Look into whether it's worth inheriting from `Common::Resource`?